lib/application.js

```diff
@@ -102,6 +103,9 @@ module.exports = class Application extends Emitter {
    */
 
   use(fn) {
+    if (this.forceAsync && fn.constructor.name !== 'AsyncFunction') {
+      throw new TypeError('middleware must be an AsyncFunction!');
+    }
```

Comment:
I'll respectfully disagree. If `forceAsync` is true getting `must be an AsyncFunction` thrown seems pretty spot on no matter if you're passing a string or a regular function IMO.

But in any case, I'll just hold of changes and conflict resolution until "dust has settled".